libs/safe/src/lib/survey/components/users.ts

```diff
@@ -43,6 +57,27 @@ export const init = (
         required: true,
       });
 
+      Survey.Serializer.addProperty('users', {
+        name: 'filterBy',
+        category: 'Users properties',
+        dependsOn: 'applications',
+        // Only makes sense to show this property if there is only one application selected
+        visibleIf: (obj: null | Question) => obj?.applications?.length === 1,
+        visibleIndex: 3,
+        choices: (obj: Question, choicesCallback: any) => {
+          if (obj.applications?.length) {
+            getApplicationRoles(obj.applications).subscribe(({ data }) => {
+              if (!data) choicesCallback([]);
+              const roles = data.rolesFromApplications.map((r) => ({
+                value: r.id,
+                text: r.title?.split(' - ')[1],
+              }));
+              choicesCallback(['', ...roles]);
```

Comment:
```suggestion
              choicesCallback([{value: null, text: ''}, ...roles]);
```
We make sure all items follow same structure to avoid any unwanted behavior in all cases